src/highlight/agent.ts

```diff
@@ -1,39 +1,59 @@
-import { Interface } from '@ethersproject/abi';
+import { TypedDataField } from '@ethersproject/abstract-signer';
 import Process from './process';
+import { BASE_DOMAIN, verifySignature } from './signatures';
+import { PostMessageRequest } from './types';
 
 export default class Agent {
   public id: string;
   public process: Process;
-  public iface: Interface;
+  public entrypoints: Record<string, Record<string, TypedDataField[]>> = {};
 
-  constructor(
-    id: string,
-    process: Process,
-    fragments: ConstructorParameters<typeof Interface>[0]
-  ) {
+  constructor(id: string, process: Process) {
     this.id = id;
     this.process = process;
-    this.iface = new Interface(fragments);
   }
 
-  invoke(data: string) {
-    const parsed = this.iface.parseTransaction({ data });
+  addEntrypoint(name: string, types: Record<string, TypedDataField[]>) {
+    this.entrypoints[name] = types;
+  }
+
+  async invoke(request: PostMessageRequest) {
+    const { entrypoint, domain, signer, signature, message } = request;
+
+    const entrypointTypes = this.entrypoints[entrypoint];
+    if (!entrypointTypes) {
+      throw new Error(`Entrypoint not found: ${entrypoint}`);
+    }
 
-    const handlerName = parsed.name;
-    const parsedArgs = parsed.args.map(arg => {
-      if (arg._isBigNumber) {
-        return arg.toBigInt();
+    const verifyingDomain = {
+      ...BASE_DOMAIN,
+      chainId: domain.chainId,
+      salt: domain.salt.toString(),
+      verifyingContract: domain.verifyingContract
```

Comment:
I think we should also have the `version` param

Reply:
`version` comes from BASE_DOMAIN. This is done so if someone signs message with different `name` or `version` it won't be valid there, because we explicitly want `highlight/0.1.0`. If someone sends us message with `highlight/1.0.0` it shouldn't be valid.

In the future if we were to add new version we would handle it there and verify accordingly.